gms/factories/src/main/java/com/linkedin/common/factory/RestHighLevelClientFactory.java

```diff
@@ -48,16 +92,67 @@ protected RestHighLevelClient createInstance() {
 
   @Nonnull
   private static RestClient loadRestHttpClient(@Nonnull String host, int port, int threadCount,
-                                               int connectionRequestTimeout) throws Exception {
+      int connectionRequestTimeout) throws Exception {
     RestClientBuilder builder = RestClient.builder(new HttpHost(host, port, "http"))
-            .setHttpClientConfigCallback(httpAsyncClientBuilder ->
-                    httpAsyncClientBuilder.setDefaultIOReactorConfig(IOReactorConfig.custom()
-                            .setIoThreadCount(threadCount).build()));
+        .setHttpClientConfigCallback(httpAsyncClientBuilder -> httpAsyncClientBuilder.setDefaultIOReactorConfig(
+            IOReactorConfig.custom().setIoThreadCount(threadCount).build()));
 
     builder.setRequestConfigCallback(requestConfigBuilder -> requestConfigBuilder.
-            setConnectionRequestTimeout(connectionRequestTimeout));
+        setConnectionRequestTimeout(connectionRequestTimeout));
 
     return builder.build();
   }
+
+  @Nonnull
+  private static RestClient loadRestHttpsClient(@Nonnull String host, int port, int threadCount,
```

Comment:
You can combine some of this with the other overload of `loadRestHttpClient` if you just have this method return the SSLContext. The difference in RestClient construction is just one extra line.